packages/manager/modules/vps/src/upscale/summary-price/summary-price.controller.js

```diff
@@ -0,0 +1,47 @@
+import includes from 'lodash/includes';
+import isNumber from 'lodash/isNumber';
+import isObject from 'lodash/isObject';
+import isString from 'lodash/isString';
+import values from 'lodash/values';
+import { PAYMENT_TYPES } from './summary-price.constants';
+import PriceService from '../price/price.service';
+
+export default class {
+  $onInit() {
+    this.validateLanguage();
+    PriceService.validatePrice(this.price);
+    this.validatePaymentType();
+
+    // the variables used in the view
+    this.v = {
+      price: PriceService.buildPriceToDisplay(this.price, this.language),
+    };
+  }
+
+  validateLanguage() {
```

Comment:
I see validateLanguage function defined in multiple places, is it possible to write once and reuse?

Reply:
It's very very linked to the component so it would be a bad idea imo